x/icacallbacks/keeper/keeper.go

```diff
@@ -102,21 +103,27 @@ func (k Keeper) CallRegisteredICACallback(ctx sdk.Context, modulePacket channelt
 		k.Logger(ctx).Info(errMsg)
 		return nil
 	}
+
 	// fetch the callback function
 	callbackHandler, err := k.GetICACallbackHandler(module)
 	if err != nil {
-		k.Logger(ctx).Info("CallRegisteredICACallback", "err", err)
-		return err
+		errMsg := fmt.Sprintf("Callback handler does not exist for module %s | err: %s", module, err.Error())
+		k.Logger(ctx).Error(errMsg)
+		return sdkerrors.Wrapf(types.ErrCallbackHandlerNotFound, errMsg)
 	}
+
 	// call the callback
 	if callbackHandler.HasICACallback(callbackData.CallbackId) {
 		// if acknowledgement is empty, then it is a timeout
 		err := callbackHandler.CallICACallback(ctx, callbackData.CallbackId, modulePacket, acknowledgement, callbackData.CallbackArgs)
 		if err != nil {
-			k.Logger(ctx).Info("CallRegisteredICACallback", "err", err)
-			return err
+			errMsg := fmt.Sprintf("Error occured while calling ICACallback (%s) | err: %s", callbackData.CallbackId, err.Error())
+			k.Logger(ctx).Error(errMsg)
+			return sdkerrors.Wrapf(types.ErrCallbackFailed, errMsg)
 		}
 	}
+	// QUESTION: Do we want to catch the case where the callback ID has not been registered?
```

Comment:
Good question -- In what scenario would a received ibc packet traverse this codepath, but not have a callback handler?

Reply:
I'm not sure. I saw that there were some blank callback Id's coming in but I think this was cause we were calling `SubmitTxEpoch(..., "", nil) ` if a few places as a stop gap before the callback was implemented (in which case, it would be updated to `SubmitTxEpoch(..., callbackId, callbackArgs)`.
Might make sense to re-evaluate once we have all the callbacks implemented in the same branch

Reply:
I think IBC transfer ack's might flow through this as well

Reply:
IBC transfer acks don't flow through here, I think this case would represent an error (`SubmitTx...` takes in a `callbackId` that has no associated function).

Reply:
IBC transfer acks don't flow through here.

`"", nil` represents ICAs that don't have an associated callback, which would reach this code path, although an unregistered callback would also reach this code path. We might want to update to

```
if callbackData.CallbackId == "" {
    return nil
} else if callbackHandler.HasICACallback(callbackData.CallbackId) {
	// if acknowledgement is empty, then it is a timeout
	err := callbackHandler.CallICACallback(ctx, callbackData.CallbackId, modulePacket, acknowledgement, callbackData.CallbackArgs)
	if err != nil {
		errMsg := fmt.Sprintf("Error occured while calling ICACallback (%s) | err: %s", callbackData.CallbackId, err.Error())
		k.Logger(ctx).Error(errMsg)
		return sdkerrors.Wrapf(types.ErrCallbackFailed, errMsg)
	}
} else {
	k.Logger(ctx).Error(fmt.Sprintf("Callback %v has no associated callback", callbackData))
}
```

Reply:
whoops, went I sent that last message I was thinking this was in `module_ibc` 

I like the change!